skbio/sequence/_iupac_sequence.py

```diff
@@ -19,6 +19,13 @@
 from ._sequence import Sequence
 
 
+def _invert_case(array):
+    # ASCII is built such that the difference between uppercase and
+    # lowercase is the 6th bit.
+    array ^= 32
```

Comment:
I'm fine with either a function or inline since this only occurs in two places. Inline might be clearer because it's obvious that it operates in place. It'll also be a little faster since there isn't the overhead of a function call (this could be called a lot of times, e.g., when processing a large FASTA file of lowercase sequences).

If we go inline, :+1: for using a constant to represent the case offset. If we go with a function, a name like `_convert_lower_to_upper` would be better since it isn't inverting the case.